File: apk-out/sources/com/nick/myrecoverytracker/MovementCaptureService.java
package com.nick.myrecoverytracker;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.app.Service;
import android.content.Intent;
import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;
import android.os.IBinder;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.core.app.NotificationCompat;
import com.google.android.gms.location.ActivityRecognition;
import com.google.android.gms.location.ActivityRecognitionClient;
import com.google.android.gms.location.ActivityTransition;
import com.google.android.gms.location.ActivityTransitionRequest;
import java.io.File;
import java.time.Instant;
import java.time.ZoneId;
import java.time.format.DateTimeFormatter;
import java.util.Locale;
import kotlin.Metadata;
import kotlin.collections.CollectionsKt;
import kotlin.io.FilesKt;
import kotlin.jvm.internal.Intrinsics;

/* compiled from: MovementCaptureService.kt */
@Metadata(d1 = {"\u0000f\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0003\n\u0002\u0010\b\n\u0002\b\u0002\n\u0002\u0010\u000e\n\u0002\b\u0002\u0018\u00002\u00020\u00012\u00020\u0002B\u0007¢\u0006\u0004\b\u0003\u0010\u0004J\b\u0010\u0014\u001a\u00020\u0015H\u0016J\b\u0010\u0016\u001a\u00020\u0015H\u0016J\u0014\u0010\u0017\u001a\u0004\u0018\u00010\u00182\b\u0010\u0019\u001a\u0004\u0018\u00010\u001aH\u0016J\u0010\u0010\u001b\u001a\u00020\u00152\u0006\u0010\u001c\u001a\u00020\u001dH\u0016J\u001a\u0010\u001e\u001a\u00020\u00152\b\u0010\u001f\u001a\u0004\u0018\u00010\b2\u0006\u0010 \u001a\u00020!H\u0016J\u0018\u0010\"\u001a\u00020\u00152\u0006\u0010#\u001a\u00020$2\u0006\u0010%\u001a\u00020$H\u0002R\u000e\u0010\u0005\u001a\u00020\u0006X\u0082.¢\u0006\u0002\n\u0000R\u0010\u0010\u0007\u001a\u0004\u0018\u00010\bX\u0082\u000e¢\u0006\u0002\n\u0000R\u000e\u0010\t\u001a\u00020\nX\u0082.¢\u0006\u0002\n\u0000R\u000e\u0010\u000b\u001a\u00020\fX\u0082.¢\u0006\u0002\n\u0000R\u0018\u0010\r\u001a\n \u000f*\u0004\u0018\u00010\u000e0\u000eX\u0082\u0004¢\u0006\u0004\n\u0002\u0010\u0010R\u0018\u0010\u0011\u001a\n \u000f*\u0004\u0018\u00010\u00120\u0012X\u0082\u0004¢\u0006\u0004\n\u0002\u0010\u0013¨\u0006&"}, d2 = {"Lcom/nick/myrecoverytracker/MovementCaptureService;", "Landroid/app/Service;", "Landroid/hardware/SensorEventListener;", "<init>", "()V", "sm", "Landroid/hardware/SensorManager;", "stepSensor", "Landroid/hardware/Sensor;", "arClient", "Lcom/google/android/gms/location/ActivityRecognitionClient;", "arPi", "Landroid/app/PendingIntent;", "zone", "Ljava/time/ZoneId;", "kotlin.jvm.PlatformType", "Ljava/time/ZoneId;", "fmtTs", "Ljava/time/format/DateTimeFormatter;", "Ljava/time/format/DateTimeFormatter;", "onCreate", "", "onDestroy", "onBind", "Landroid/os/IBinder;", "intent", "Landroid/content/Intent;", "onSensorChanged", NotificationCompat.CATEGORY_EVENT, "Landroid/hardware/SensorEvent;", "onAccuracyChanged", "sensor", "accuracy", "", "append", "name", "", "line", "app_debug"}, k = 1, mv = {2, 0, 0}, xi = ConstraintLayout.LayoutParams.Table.LAYOUT_CONSTRAINT_VERTICAL_CHAINSTYLE)
/* loaded from: classes3.dex */
public final class MovementCaptureService extends Service implements SensorEventListener {
    private ActivityRecognitionClient arClient;
    private PendingIntent arPi;
    private SensorManager sm;
    private Sensor stepSensor;
    private final ZoneId zone = ZoneId.systemDefault();
    private final DateTimeFormatter fmtTs = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss", Locale.US);

    @Override // android.app.Service
    public void onCreate() {
        super.onCreate();
        Object systemService = getSystemService("notification");
        Intrinsics.checkNotNull(systemService, "null cannot be cast to non-null type android.app.NotificationManager");
        NotificationManager nm = (NotificationManager) systemService;
        if (nm.getNotificationChannel("movement_fg") == null) {
            nm.createNotificationChannel(new NotificationChannel("movement_fg", "Movement", 1));
        }
        Notification n = new Notification.Builder(this, "movement_fg").setContentTitle("Movement capture active").setSmallIcon(android.R.drawable.stat_notify_sync_noanim).setOngoing(true).build();
        Intrinsics.checkNotNullExpressionValue(n, "build(...)");
        startForeground(Notifier.ID_ONBOARDING, n);
        Object systemService2 = getSystemService("sensor");
        Intrinsics.checkNotNull(systemService2, "null cannot be cast to non-null type android.hardware.SensorManager");
        this.sm = (SensorManager) systemService2;
        SensorManager sensorManager = this.sm;
        PendingIntent pendingIntent = null;
        if (sensorManager == null) {
            Intrinsics.throwUninitializedPropertyAccessException("sm");
            sensorManager = null;
        }
        this.stepSensor = sensorManager.getDefaultSensor(19);
        Sensor sensor = this.stepSensor;
        if (sensor != null) {
            SensorManager sensorManager2 = this.sm;
            if (sensorManager2 == null) {
                Intrinsics.throwUninitializedPropertyAccessException("sm");
                sensorManager2 = null;
            }
            sensorManager2.registerListener(this, sensor, 3);
        }
        this.arClient = ActivityRecognition.getClient(this);
        Intent intent = new Intent(this, (Class<?>) MovementARReceiver.class).setAction("mrt.AR");
        Intrinsics.checkNotNullExpressionValue(intent, "setAction(...)");
        this.arPi = PendingIntent.getBroadcast(this, 100, intent, 201326592);
        ActivityTransitionRequest req = new ActivityTransitionRequest(CollectionsKt.listOf((Object[]) new ActivityTransition[]{new ActivityTransition.Builder().setActivityType(3).setActivityTransition(0).build(), new ActivityTransition.Builder().setActivityType(7).setActivityTransition(0).build(), new ActivityTransition.Builder().setActivityType(2).setActivityTransition(0).build(), new ActivityTransition.Builder().setActivityType(8).setActivityTransition(0).build(), new ActivityTransition.Builder().setActivityType(0).setActivityTransition(0).build(), new ActivityTransition.Builder().setActivityType(1).setActivityTransition(0).build()}));
        ActivityRecognitionClient activityRecognitionClient = this.arClient;
        if (activityRecognitionClient == null) {
            Intrinsics.throwUninitializedPropertyAccessException("arClient");
            activityRecognitionClient = null;
        }
        PendingIntent pendingIntent2 = this.arPi;
        if (pendingIntent2 == null) {
            Intrinsics.throwUninitializedPropertyAccessException("arPi");
        } else {
            pendingIntent = pendingIntent2;
        }
        activityRecognitionClient.requestActivityTransitionUpdates(req, pendingIntent);
    }

    @Override // android.app.Service
    public void onDestroy() {
        PendingIntent pendingIntent = null;
        try {
            SensorManager sensorManager = this.sm;
            if (sensorManager == null) {
                Intrinsics.throwUninitializedPropertyAccessException("sm");
                sensorManager = null;
            }
            sensorManager.unregisterListener(this);
        } catch (Throwable th) {
        }
        try {
            ActivityRecognitionClient activityRecognitionClient = this.arClient;
            if (activityRecognitionClient == null) {
                Intrinsics.throwUninitializedPropertyAccessException("arClient");
                activityRecognitionClient = null;
            }
            PendingIntent pendingIntent2 = this.arPi;
            if (pendingIntent2 == null) {
                Intrinsics.throwUninitializedPropertyAccessException("arPi");
            } else {
                pendingIntent = pendingIntent2;
            }
            activityRecognitionClient.removeActivityTransitionUpdates(pendingIntent);
        } catch (Throwable th2) {
        }
        super.onDestroy();
    }

    @Override // android.app.Service
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override // android.hardware.SensorEventListener
    public void onSensorChanged(SensorEvent event) {
        Intrinsics.checkNotNullParameter(event, "event");
        if (event.sensor.getType() != 19) {
            return;
        }
        String ts = Instant.ofEpochMilli(System.currentTimeMillis()).atZone(this.zone).format(this.fmtTs);
        String line = ts + ",step," + event.values[0];
        append("movement_log.csv", line);
    }

    @Override // android.hardware.SensorEventListener
    public void onAccuracyChanged(Sensor sensor, int accuracy) {
    }

    private final void append(String name, String line) {
        File f = new File(getFilesDir(), name);
        File parentFile = f.getParentFile();
        if (parentFile != null) {
            parentFile.mkdirs();
        }
        FilesKt.appendText$default(f, line + "\n", null, 2, null);
    }
}

File: app/src/main/java/com/nick/myrecoverytracker/MovementCaptureService.kt
package com.nick.myrecoverytracker

import android.app.*
import android.content.*
import android.hardware.Sensor
import android.hardware.SensorEvent
import android.hardware.SensorEventListener
import android.hardware.SensorManager
import android.os.IBinder
import com.google.android.gms.location.*
import java.io.File
import java.time.Instant
import java.time.ZoneId
import java.time.format.DateTimeFormatter
import java.util.Locale
import kotlin.math.sqrt

class MovementCaptureService : Service(), SensorEventListener {
    private lateinit var sm: SensorManager
    private var stepSensor: Sensor? = null
    private var accSensor: Sensor? = null
    private lateinit var arClient: ActivityRecognitionClient
    private lateinit var arPi: PendingIntent
    private val zone = ZoneId.systemDefault()
    private val fmtTs = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss", Locale.US)

    // Accelerometer baseline tracking
    private var lastAccTs: Long = 0L
    private var baselineSet = false
    private var baseX = 0f
    private var baseY = 0f
    private var baseZ = 0f

    companion object {
        private var instance: MovementCaptureService? = null

        fun writeActive() {
            instance?.logActivity("1.000000")
        }

        fun writeStill() {
            instance?.logActivity("0.000000")
        }
    }

    override fun onCreate() {
        super.onCreate()
        instance = this

        val chId = "movement_fg"
        val nm = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
        if (nm.getNotificationChannel(chId) == null) {
            nm.createNotificationChannel(NotificationChannel(chId, "Movement", NotificationManager.IMPORTANCE_MIN))
        }
        val n = Notification.Builder(this, chId)
            .setContentTitle("Movement capture active")
            .setSmallIcon(android.R.drawable.stat_notify_sync_noanim)
            .setOngoing(true)
            .build()
        startForeground(2001, n)

        sm = getSystemService(SENSOR_SERVICE) as SensorManager

        // Register step counter
        stepSensor = sm.getDefaultSensor(Sensor.TYPE_STEP_COUNTER)
        stepSensor?.let { sm.registerListener(this, it, SensorManager.SENSOR_DELAY_NORMAL) }

        // Register accelerometer
        accSensor = sm.getDefaultSensor(Sensor.TYPE_ACCELEROMETER)
        accSensor?.let { sm.registerListener(this, it, SensorManager.SENSOR_DELAY_NORMAL) }

        // Setup Activity Recognition
        arClient = ActivityRecognition.getClient(this)
        val intent = Intent(this, MovementARReceiver::class.java).setAction("mrt.AR")
        arPi = PendingIntent.getBroadcast(this, 2001, intent, PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)
        val req = ActivityTransitionRequest(listOf(
            ActivityTransition.Builder().setActivityType(DetectedActivity.STILL).setActivityTransition(ActivityTransition.ACTIVITY_TRANSITION_ENTER).build(),
            ActivityTransition.Builder().setActivityType(DetectedActivity.WALKING).setActivityTransition(ActivityTransition.ACTIVITY_TRANSITION_ENTER).build(),
            ActivityTransition.Builder().setActivityType(DetectedActivity.ON_FOOT).setActivityTransition(ActivityTransition.ACTIVITY_TRANSITION_ENTER).build(),
            ActivityTransition.Builder().setActivityType(DetectedActivity.RUNNING).setActivityTransition(ActivityTransition.ACTIVITY_TRANSITION_ENTER).build(),
            ActivityTransition.Builder().setActivityType(DetectedActivity.IN_VEHICLE).setActivityTransition(ActivityTransition.ACTIVITY_TRANSITION_ENTER).build(),
            ActivityTransition.Builder().setActivityType(DetectedActivity.ON_BICYCLE).setActivityTransition(ActivityTransition.ACTIVITY_TRANSITION_ENTER).build()
        ))
        arClient.requestActivityTransitionUpdates(req, arPi)
    }

    override fun onDestroy() {
        try { sm.unregisterListener(this) } catch (_: Throwable) {}
        try { arClient.removeActivityTransitionUpdates(arPi) } catch (_: Throwable) {}
        instance = null
        super.onDestroy()
    }

    override fun onBind(intent: Intent?): IBinder? = null

    override fun onSensorChanged(event: SensorEvent) {
        when (event.sensor.type) {
            Sensor.TYPE_STEP_COUNTER -> {
                val ts = Instant.ofEpochMilli(System.currentTimeMillis()).atZone(zone).format(fmtTs)
                val line = "$ts,step,${event.values[0]}"
                append("movement_log.csv", line)
            }
            Sensor.TYPE_ACCELEROMETER -> {
                handleAccelerometer(event)
            }
        }
    }

    private fun handleAccelerometer(e: SensorEvent) {
        if (!baselineSet) {
            baseX = e.values[0]
            baseY = e.values[1]
            baseZ = e.values[2]
            baselineSet = true
            lastAccTs = 0L
            return
        }
        val dx = e.values[0] - baseX
        val dy = e.values[1] - baseY
        val dz = e.values[2] - baseZ
        val mag = sqrt((dx*dx + dy*dy + dz*dz).toDouble())
        val nowMs = System.currentTimeMillis()
        if (nowMs - lastAccTs >= 10000L) {
            lastAccTs = nowMs
            writeMagnitude(mag)
        }
    }

    private fun writeMagnitude(mag: Double) {
        val ts = Instant.ofEpochMilli(System.currentTimeMillis()).atZone(zone).format(fmtTs)
        val v = String.format(Locale.US, "%.6f", mag)
        append("movement_log.csv", "$ts,$v")
    }

    private fun logActivity(value: String) {
        val ts = Instant.ofEpochMilli(System.currentTimeMillis()).atZone(zone).format(fmtTs)
        append("movement_log.csv", "$ts,$value")
    }

    override fun onAccuracyChanged(sensor: Sensor?, accuracy: Int) {}

    private fun append(name: String, line: String) {
        val f = File(filesDir, name)
        f.parentFile?.mkdirs()
        f.appendText(line + "\n")
    }
}